sdk/communication/Azure.Communication.Rooms/src/RoomsClient.cs

```diff
@@ -0,0 +1,511 @@
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="tokenCredential"></param>
+        /// <param name="options"></param>
+        public RoomsClient(Uri endpoint, TokenCredential tokenCredential, RoomsClientOptions options = default)
+            : this(
+                Argument.CheckNotNull(endpoint, nameof(endpoint)).AbsoluteUri,
+                Argument.CheckNotNull(tokenCredential, nameof(tokenCredential)),
+                options ?? new RoomsClientOptions())
+        { }
+
+        #endregion
+
+        #region private constructors
+
+        private RoomsClient(ConnectionString connectionString, RoomsClientOptions options)
+            : this(connectionString.GetRequired("endpoint"), options.BuildHttpPipeline(connectionString), options)
+        { }
+
+        private RoomsClient(string endpoint, TokenCredential tokenCredential, RoomsClientOptions options)
+            : this(endpoint, options.BuildHttpPipeline(tokenCredential), options)
+        { }
+
+        private RoomsClient(string endpoint, AzureKeyCredential keyCredential, RoomsClientOptions options)
+            : this(endpoint, options.BuildHttpPipeline(keyCredential), options)
+        { }
+
+        private RoomsClient(string endpoint, HttpPipeline httpPipeline, RoomsClientOptions options)
+        {
+            _clientDiagnostics = new ClientDiagnostics(options);
+            RoomsServiceClient = new RoomsRestClient(_clientDiagnostics, httpPipeline, endpoint, options.ApiVersion);
+        }
+        #endregion
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
+        /// </summary>
+        protected RoomsClient()
+        {
+            _clientDiagnostics = null!;
+            RoomsServiceClient = null!;
+        }
+
+        /// <summary>
+        /// Creates a new room asynchronously.
+        /// </summary>
+        /// <param name="validFrom"></param>
+        /// <param name="validUntil"></param>
+        /// <param name="roomJoinPolicy"></param>
+        /// <param name="participants"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        public virtual async Task<Response<CommunicationRoom>> CreateRoomAsync(DateTimeOffset? validFrom = default, DateTimeOffset? validUntil = default, RoomJoinPolicy? roomJoinPolicy = default, IEnumerable<RoomParticipant> participants = default, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(RoomsClient)}.{nameof(CreateRoom)}");
+            scope.Start();
+            try
+            {
+                Guid repeatabilityRequestId = Guid.NewGuid();
+                DateTimeOffset repeatabilityFirstSent = DateTimeOffset.UtcNow;
```

Comment:
Could we handle this with a pipeline policy?  It feels more like plumbing than service specific logic.

Reply:
I am not understanding the pipeline policy here. Sample reference will help.
This is the way I see other communication services are defining parameters for date, participants. Please refer to Chat.

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Pipeline.md#azurecore-pipeline-samples

Reply:
We only need the headers in a create call, not every single type of HTTP request...so do you still recommend using a pipeline if this is the case?

Reply:
@tg-msft tagging so we can close the loop on this question.

Reply:
We could have the policy check and only apply them on messages with `PUT`, but it's fine to leave this for now if you think it'll be rare.

Reply:
We will leave it as such for now as it is only one call that needs the header.